quick-start/demo-function/Dockerfile

```diff
@@ -0,0 +1,26 @@
+# Based on vanilla Dockerfile to build a Golang image from:
+# https://docs.aws.amazon.com/lambda/latest/dg/go-image.html
+# The only significant modification is to copy the extension binary into the image.
+FROM public.ecr.aws/lambda/provided:al2 as build
```

Comment:
Is there a hard requirement to use this Docker image for builds? This Dockerfile uses a builder pattern to produce a binary, so does it make more sense to use the Golang Docker image instead?

I thing the Golang docker image might be a bit better for maintenance because right now `yum` is just installing latest version of Golang, but the Docker image would be versioned with a specific Golang installation.

Reply:
Yeah I think that's a worthwhile improvement. As per the comment, this file is almost entirely just lifted from AWS docs because the only thing it's really trying to demonstrate is that the only additional step you need for extensions is the copy at the bottom (the rest is setting up the function packaging), but it does make more sense to use a versioned image for the builder.

Reply:
It's up to you, but maybe we should be more explicit with the version of Golang on `yum install` if we're going to keep this image.